students_db/src/main/java/application/dao_interfaces/GroupDisciplineDao.java

```diff
@@ -6,7 +6,7 @@
 
 public interface GroupDisciplineDao {
 
-    void insertGroupDiscipline(GroupDiscipline groupDiscipline);
+    void insert(GroupDiscipline groupDiscipline);
```

Comment:
you can remove it - you interface should be generic - don't copy all methods with needed type